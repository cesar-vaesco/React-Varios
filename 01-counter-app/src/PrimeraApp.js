import React from 'react';
import PropTypes from 'prop-types';


const PrimeraApp = ({ saludo, subtitulo, mensaje }) => {


    return (
        <>
            {/* <pre>{ JSON.stringify (persona, null, 3)}</pre> */}
            <h1>{saludo}</h1>
            <h2> {subtitulo} </h2>
            <p>Mi primer Aplicaci√≥n</p>
            <p>{ mensaje}</p>
        </>
    );

}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    mensaje: 'Conociendo las defaultProps de React',
    subtitulo: 'Soy un subtitulo'
}

export default PrimeraApp;
