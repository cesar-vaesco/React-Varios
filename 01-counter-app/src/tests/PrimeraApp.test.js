import '@testing-library/jest-dom';

// eslint-disable-next-line no-unused-vars
import React from 'react';
// eslint-disable-next-line no-unused-vars
import PrimeraApp from '../PrimeraApp';

import { shallow } from 'enzyme';

describe('Pruebas en PrimeraApp', () => {

    // test('Debe de mostrar el mensaje: Hola, Soy Goku', () => {
    //     const saludo = 'Hola, Soy Goku';
    //     const { getByText } = render(<PrimeraApp saludo={saludo} />);
    //     expect(getByText(saludo)).toBeInTheDocument();
    // });

    test('Debe de mostrer <PrimeraApp/> correctamente', () => {

        const saludo = 'Hola, Soy Goku';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar el subtitulo enviado por props', () => {
        // 1.- Inicializaci√≥n
        const saludo = 'Hola, Soy Goku';
        const subTitulo = 'Soy un subtitulo';
        //  2.- estimulo
        const wrapper = shallow(<PrimeraApp
            saludo={saludo}
            subtitulo={subTitulo}
        />
        );
        //const textoParrafo = wrapper.find('p').text();
        const textoParrafo = wrapper.find('h2').text().trim();
        console.log(textoParrafo);
        // 3.- Observar el comportamiento
        expect(textoParrafo).toBe(subTitulo);
    });
});
