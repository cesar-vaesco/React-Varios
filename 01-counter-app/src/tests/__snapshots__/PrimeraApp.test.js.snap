// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp Debe de mostrer <PrimeraApp/> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, Soy Goku
  </h1>
  <h2>
     
    Soy un subtitulo
     
  </h2>
  <p>
    Mi primer Aplicaci√≥n
  </p>
  <p>
    Conociendo las defaultProps de React
  </p>
</Fragment>
`;
